test/long/org/apache/cassandra/dht/tokenallocator/TokenAllocatorTestBase.java

```diff
@@ -111,7 +121,8 @@ public void random(Map<Token, Unit> map, TestReplicationStrategy rs,
             int tokens = tc.tokenCount(perUnitCount, rand);
             for (int j = 0; j < tokens; j++)
             {
-                map.put(partitioner.getRandomToken(rand), unit);
+                Token token = partitioner.getRandomToken(rand);
+                map.put(partitioner.split(partitioner.getMinimumToken(), token, skewFactor), unit);
```

Comment:
this is a no-op when skewFactor is 1. Otherwise, this forces all the randomly chosen tokens to be in a subset of the ring.